### PR TITLE
added points integration to KCom2 Checkout as per #6218

### PR DESCRIPTION
**Pull request by BrewsterNZ**
I've had to manually move changes over from one source branch to another so let me know if anything is missing.

### CODE REVIEW

File: src/main/marketplace/apps/KCommerce2/APP-INF/mappings.js [hunk 1]
```diff
@@ -40,7 +40,9 @@ var cartMapping = controllerMappings
         .addMethod('POST', 'setCartItem', 'quantity')
         .addMethod('POST', 'removeCartItem', 'removeLineId')
         .addMethod('POST', 'setCartItemQuantity', 'newQuantity')
-        .addMethod('POST', 'checkout', 'processCartId')
+        .addMethod('POST', 'setCartPointsAllocation', 'newPointsAllocation')
+        .addMethod('POST', 'checkout', 'processCart')
```
Comment (bradmac) [inline src/main/marketplace/apps/KCommerce2/APP-INF/mappings.js]: This looks incorrect. The html is using an input called processCartId, not processCart.

I suggest this line is reverted to the original

File: src/main/marketplace/apps/KCommerce2/website/theme/apps/KCommerce2/components/ecomCheckoutFormPointsOnlyComponent.html [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        #set( $reward = $services.pointsManager.findPointsBucket($pointsBucket) )
+        #set( $pointsBalance = $services.pointsManager.pointsBalance($reward, $user) )
+    #end
+
+    #if( $pointsBalance < $checkoutItems.totalCost)
+        <p>You don't have enough points to purchase these items</p>
+    #else
+
+
+    <script>
```
Comment (bradmac) [inline src/main/marketplace/apps/KCommerce2/website/theme/apps/KCommerce2/components/ecomCheckoutFormPointsOnlyComponent.html]: A lot of this looks very similar with the standard checkout component.

Can you look at using a velocity include, so the code isnt duplicated? See ecommerceMacros.html as an example

File: src/main/marketplace/apps/KCommerce2/APP-INF/mappings.js (hunk 1 above)
Comment (BrewsterNZ) [reply src/main/marketplace/apps/KCommerce2/APP-INF/mappings.js]: Both the std checkout and the points checkout supply a processCart Id.
They now also supply a hidden field to distinguish themselves so the correct checkout function can run.

File: src/main/marketplace/apps/KCommerce2/website/theme/apps/KCommerce2/components/ecomCheckoutFormPointsOnlyComponent.html (hunk 2 above)
Comment (BrewsterNZ) [reply src/main/marketplace/apps/KCommerce2/website/theme/apps/KCommerce2/components/ecomCheckoutFormPointsOnlyComponent.html]: Yeah I was worried about this also.
Will do in a subsequent PR.

**Pull request merged**